src/grid/onedgrid.py

```diff
@@ -114,6 +114,294 @@ def GaussChebyshev(npoints):
+
+
+    Parameters
+    ----------
+    npoints : int
+        Number of points in the grid.
+
+    Returns
+    -------
+    OneDGrid
+        A 1D grid instance.
+
+    """
+    idx = np.arange(npoints) + 1
+    points = idx / (npoints + 1)
+
+    weights = np.zeros(npoints)
+
+    index_m = np.arange(npoints) + 1
+
+    for i in range(0, npoints):
+        elements = np.zeros(npoints)
+        elements = np.sin(index_m * np.pi * points[i])
+        elements = elements * (1 - np.cos(index_m * np.pi)) / (index_m * np.pi)
+
+        weights[i] = (2 / (npoints + 1)) * np.sum(elements)
+
+    return OneDGrid(points, weights, (0, 1))
+
+
+def RectangleRuleSine(npoints):
+    r"""Generate 1D grid on (0:1) interval based on rectangle rule.
+
+    The fundamental definition of this quadrature is:
+
+    .. math::
+        \int_{0}^{1} f(x) dx \approx \sum_{i=1}^n w_i f(x_i)
+
+    Where
+
+    .. math::
+        x_i = \frac{2 i - 1}{2 n}
+
+    And the weights
+
+    .. math::
+        w_i = \frac{2}{n^2 \pi} \sin(n\pi x_i) \sin^2(n\pi /2)
+            + \frac{4}{n \pi}\sum_{m=1}^{n-1}
+            \frac{\sin(m \pi x_i)\sin^2(m\pi /2)}
+                {m}
+
+    Parameters
+    ----------
+    npoints : int
+        Number of points in the grid.
+
+    Returns
+    -------
+    OneDGrid
+        A 1D grid instance.
+
+    """
+    idx = np.arange(npoints) + 1
+    points = (2 * idx - 1) / (2 * npoints)
+
+    weights = np.zeros(npoints)
+
+    index_m = np.arange(npoints - 1) + 1
+
+    for i in range(0, npoints):
+        elements = np.zeros(npoints - 1)
+        elements = np.sin(index_m * np.pi * points[i])
+        elements *= np.sin(index_m * np.pi / 2) ** 2
+        elements /= index_m
+
+        weights[i] = (4 / (npoints * np.pi)) * np.sum(elements)
+
+        weights[i] += (
+            (2 / (npoints * np.pi ** 2))
+            * np.sin(npoints * np.pi * points[i])
+            * np.sin(npoints * np.pi / 2) ** 2
+        )
+
+    return OneDGrid(points, weights, (0, 1))
+
+
+def TanhSinh(npoints, delta):
+    r"""Generate 1D grid on [-1,1] interval based on Tanh-Sinh rule.
+
+    The fundamental definition is:
+
+    .. math::
+        \int_{-1}^{1} f(x) dx \approx
+        \sum_{i=-\frac{1}{2}(n-1)}^{\frac{1}{2}(n-1)} w_i f(x_i)
+
+    Where
+
+    .. math::
+        x_i = \tanh\left( \frac{\pi}{2} \sinh(i\delta) \right)
+
+    And the weights
+
+    .. math::
+        w_i = \frac{\frac{\pi}{2}\delta \cosh(i\delta)}
+        {\cosh^2(\frac{\pi}{2}\sinh(i\delta))}
+
+
+    Parameters
+    ----------
+    npoints : int
+        Number of points in the grid, this value must be odd.
+
+    delta : float
+        This values is a parameter :math:`\delta`, is related with the size.
+
+    Returns
+    -------
+    OneDGrid
+        A 1D grid instance.
+    """
+    if npoints % 2 == 0:
+        raise ValueError("npoints must be odd, given {npoints}")
+
+    jmin = (int)(1 - npoints) / 2
```

Comment:
Everything else looks good to me except that I am not sure what ``(int)`` does here.

Reply:
Thanks, well I wrote (int) because I was thinking in python 3.6, in python 2.7 it's not necessary, sorry was my mistake, jmin is a counter.